.github/workflows/alembic-causal-ai.yml

```
@@ -0,0 +1,173 @@
+          - events_only
+          - causal_analysis_only
+          - data_collection_only
+        default: 'full'
+
+jobs:
+  alembic-orchestration:
+    runs-on: ubuntu-latest
+    timeout-minutes: 25  # Ensure it completes before next run
+    
+    steps:
+      - uses: actions/checkout@v3
```

actionlint (1.7.7)

the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)